packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java

```diff
@@ -110,12 +117,113 @@ public void onMethodCall(MethodCall call, Result result) {
       case "stopListeningAuthState":
         handleStopListeningAuthState(call, result);
         break;
+      case "verifyPhoneNumber":
+        handleVerifyPhoneNumber(call, result);
+        break;
+      case "signInWithPhoneNumber":
+        handleSignInWithPhoneNumber(call, result);
+        break;
       default:
         result.notImplemented();
         break;
     }
   }
 
+  private void handleSignInWithPhoneNumber(MethodCall call, Result result) {
+    Map<String, String> arguments = (Map<String, String>) call.arguments;
+    String verificationId = arguments.get("verificationId");
```

Comment:
If `call.arguments` is a map, I believe you can use `call.arguments("verificationId")` instead. Then you could avoid the casting above.

Reply:
The cast to `Map<String, String>` would still be necessary so I'll leave it as is.